src/Illuminate/Foundation/Console/RouteCacheCommand.php

```diff
@@ -78,10 +79,33 @@ public function handle()
      */
     protected function getFreshApplicationRoutes()
     {
-        return tap($this->getFreshApplication()['router']->getRoutes(), function ($routes) {
+        $application = $this->getFreshApplication();
+
+        $routes = tap($application['router']->getRoutes(), function ($routes) {
             $routes->refreshNameLookups();
             $routes->refreshActionLookups();
         });
+
+        if($this->hasOctane($application)) {
+            $warning = 'Running route:cache while running swoole decreases memory performance';
+            $this->components->warn($warning);
+            collect($application->octane->getRoutes())
+                ->each( function ($callback, $signature) use ($application) {
+                    list($method, $uri) = explode('/',$signature, 2);
+                    $application['router']->addRoute($method, $uri, $callback);
+                });
+        }
+
+        return $routes;
+    }
+
+    /**
+     * @param Application $application
+     * @return bool
+     */
+    protected function hasOctane(Application $application): bool
+    {
+        return $application?->octane !== null;
```

Comment:
Isn't this adding null-safety to $application (which cannot be null or the interpreter will raise a TypeError) and not to the Application@$octane property?

Reply:
yeah it did i removed the method and changed it for the has method on the application context.